models.py

```diff
@@ -25,6 +26,8 @@ class Machine(ModelBase):
     machineuid = CharField(unique=True)
     machinename = CharField()
     status = IntegerField(null=True)
+    costperminute = FloatField()
```

Comment:
I would suggest storing financial values as an integer (multiplied by 100), to avoid any potential for rounding errors in arithmetic. Might be worth using an unsigned integer column as an easy way of preventing storing negative values.

Reply:
The database migration shows the column definitions allowing `null` values in the `costperminute` and `costminimum` columns, but these fields aren't configured to allow nulls. To reduce the amount of business logic dealing with "free" machines, you could disallow nulls and just use a zero value.